Blockchain config docs

jrainville opened the pull request:
Wanted to add configs for the new accounts, but found there was no blockchain config at all...

Here it is.

PascalPrecht commented on source/docs/blockchain_basic_configuration.md:
```diff
@@ -0,0 +1,66 @@
+title: Blockchain configuration
+---
+
+When in development phase, it is necessary to use a local Ethereum node to deploy our contracts to, either using a simulator or client like Geth or Parity.
```
Theoretically, having a node set up for Embark is actually optional. So "necessary" might not be the right wording here. How about something like this:

> If we decide to build a decentralized application that should be powered by a blockchain, we need to tell Embark what node it should connect to, to deploy our application accordingly. This can be either a [blockchain simulator]() (link to simulator section in running apps guide), or a real node like [Geth]() or [Parity]().
>
> In this guide we'll explore how to configure the blockchain node we want Embark to connect to.

PascalPrecht commented on source/docs/blockchain_basic_configuration.md:
```diff
@@ -0,0 +1,66 @@
+
+To set this node, Embark looks at `blockchain.js`, which by default is at `config/blockchain.js`.
+
+It is a complex file, but Embark, when using the `embark new` and `embark demo` command, creates a default `blockchain.js` file for you that contains all the basic parameters for you to start a Geth node or a Ganache simulator.
```
Tiny tip: Generally it's better to make things sounds positive instead of negative. So instead of telling people that the file is complex, let's just focus on that Embark luckily comes with decent defaults out of the box, so we don't have to worry too much.

So maybe something like

> Embark offers a lot of configuration options and most of them already come with a decent default so we can start right away.

Can definitely worded better..

PascalPrecht commented on source/docs/blockchain_basic_configuration.md:
```diff
@@ -0,0 +1,66 @@
+}
+</code></pre>
+
+As with `contracts.js`, this config contains environments that help configuring certain parameters differently depending of the environment. You can read more about [environments here](https://embark.status.im/docs/environments.html).
```
Instead of mentioning `contracts.js`, which potentially can be named something totally different if the user wants to, how about we rather refer to the guide on configuring contracts?

Something like:

> Similar to [configuring Smart Contracts](), this configuration file contains [environments]() that help [...]

PascalPrecht commented on source/docs/blockchain_basic_configuration.md:
```diff
@@ -0,0 +1,66 @@
+- **networkId**: Used when `networkType` is set as `custom`. [List of known network ids](https://github.com/ethereumbook/ethereumbook/blob/3e8cf74eb935d4be495f4306b73de027af95fd97/contrib/devp2p-protocol.asciidoc#known-current-network-ids)
+- **isDev**: Whether or not to use the development mode of the Node
+  - This is a special mode where the node uses a development account as defaultAccount. This account is already funded and transactions are faster
+  - It is recommended to start by using `isDev: true` for you project, as it is faster and safer
```
This `isDev` stuff would make a great section within this guide that we could link people to if needed.

PascalPrecht commented on source/docs/blockchain_privatenet_configuration.md:
```diff
@@ -0,0 +1,43 @@
+
+### Parameter descriptions
+
+- **isDev**: You need to set `isDev` to false to use enable any network that isn't the development one
```
As mentioned in the other comment, it'd be great to have a section to link to that explains when one wants to use `isDev` and what it means.

jrainville commented:
@PascalPrecht I addressed most of the comments. The ones I didn't address are mostly the "big ones" like creating sections for each account types as I don't think there's much t say about those, but if you feel like it I'll let you have fun 😉

PascalPrecht commented:
@jrainville this is much better now!! Thank you.

I'll have another go over remaining guides in the docs to fine-tune some wording. Will do that in a separate PR.

Great job! 👏
